openmc/surface.py

```diff
@@ -94,19 +94,18 @@ def __repr__(self):
 
         return string
 
-    def __hash__(self):
-        coeffs = list(sorted(self.coefficients.items()))
-        coeff_tup = tuple(j for i in coeffs for j in i)
-        return hash((self.type, self.boundary_type) + coeff_tup)
-
-    def __eq__(self, other):
-        if isinstance(other, type(self)):
-            return hash(self) == hash(other)
-        else:
-            return False
+    def _key(self):
+        string = '{0: <16}{1}{2}\n'.format('\tType', '=\t', self._type)
+        string += '{0: <16}{1}{2}\n'.format('\tBoundary', '=\t', self._boundary_type)
+
+        coefficients = '{0: <16}'.format('\tCoefficients') + '\n'
 
-    def __ne__(self, other):
-        return not self == other
+        for coeff, value in list(sorted(self.coefficients.items())):
+            coefficients += '{0: <16}{1}{2}\n'.format(coeff, '=\t', value)
+
+        string += coefficients
+
+        return string
```

Comment:
If we make this public, something like `Surface.description`, we can use this as the key in the dictionary from the previous implementation. This gives us the one-pass approach from before, without relying on `__hash__` for Surfaces. My bad for not catching that before...

Reply:
Ah, good call. I don't know why I didn't think of that.